controllers/department.js

```diff
@@ -1,31 +1,65 @@
-const Department = require("../models/department");
+const DepartmentService = require("../services/department");
+const EmployeeService = require("../services/employee");
+const departmentValidator = require("../validations/department");
+const ClientError = require("../errors");
 
-class departmentController {
-  async getDepartments(request, response) {
+class DepartmentController {
+  async createNewDepartment(request, response, next) {
     try {
-      const departments = await Department.find();
+      const { error } = departmentValidator(request.body);
+      if (error) {
+        throw ClientError.BadRequest(error.details[0].message);
+      }
+      const department = await DepartmentService.createNewDepartment(request.body, request.files.picture);
+      return response.status(200).json({ department });
+    } catch (error) {
+      next(error);
+    }
+  }
+  async getDepartments(request, response, next) {
+    try {
+      const departments = await DepartmentService.getDepartments();
       return response.status(200).json({ departments });
-    } catch (e) {
-      console.log(e);
-      return response.status(500).json({ message: "Internal Server Error: Could not fulfil your request." });
+    } catch (error) {
+      next(error);
     }
   }
-  async postNewDepartment(request, response) {
+  async getDepartmentByID(request, response, next) {
     try {
-    } catch (e) {}
+      const departmentByID = await DepartmentService.getDepartmentByID(request.params.id);
+      return response.status(200).json({ departmentByID });
+    } catch (error) {
+      next(error);
+    }
   }
-  async getDepartmentByID(request, response) {
+  async getEmployeesByDepartmentID(request, response, next) {
     try {
-    } catch (e) {}
+      const employees = await EmployeeService.getEmployeesByDepartmentID(request.params.id);
```

Comment:
please, validate id. what if I pass json in it? and - what if everything ok and no such id in DB?